compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -256,6 +256,17 @@ object Types {
       case _ => false
     }
 
+    /** Is this an effect type? */
+    def isEffect(implicit ctx: Context): Boolean = this match {
+      case tp: TypeRef => tp.stableInRunSymbol.is(Effect)
+      case tp: TypeProxy => tp.superType.isEffect
+      case tp: MethodType => tp.resType.isEffect
+      case tp: PolyType => tp.resType.isEffect
+      case AndType(tp1, tp2) => tp1.isEffect && tp2.isEffect
```

Comment:
Given that we check that both types are effects in an `And` type, we only need to ask `tp1.isEffect`. Same for `Or`.